src/add_command.rs

```diff
@@ -32,11 +52,28 @@ impl AddCommand {
             .append(true)
             .create(true)
             .open(self.config.file_path.clone())
-            .map_err(|open_error| AddCommandError::CannotOpenFile { file_path: self.config.file_path.clone(), open_error: open_error.kind() })?;
+            .map_err(|open_error| AddCommandError::CannotOpenFile { file_path: self.config.file_path.clone(), open_error: open_error })?;
```

Comment:
can we use `From` implementation, so this mapping would be automatic by `?` operator?

Reply:
ok, guess we cannot, since same error type (io error) may map into two different errors (open error / write error) depending on when error happens